custom/icds_reports/tests/test_views_first_days_of_month.py

```diff
@@ -0,0 +1,595 @@
+        with mock.patch('custom.icds_reports.views.datetime', LastDayOfJune):
+            response_june = self.view(request, step=u'chart', domain=u'icds-test', location_id=u'b1')
+        with mock.patch('custom.icds_reports.views.datetime', SecondDayOfJuly):
+            response_first_of_july = self.view(request, step=u'chart', domain=u'icds-test', location_id=u'b1')
+
+        self.assertDictEqual(json.loads(response_june.content), json.loads(response_first_of_july.content))
+
+    def test_map_third_day_of_month(self):
+        if self.factory is None:
+            return
+        working_reverse = reverse(self.url, kwargs={'domain': 'icds-test', 'step': 'map'})
+        request = self.factory.get(working_reverse)
+        request.user = self.user
+        with mock.patch('custom.icds_reports.views.datetime', LastDayOfJune):
+            response_june = self.view(request, step=u'map', domain=u'icds-test')
+        with mock.patch('custom.icds_reports.views.datetime', ThirdDayOfJuly):
+            response_first_of_july = self.view(request, step=u'map', domain=u'icds-test')
+
+        june_data = json.loads(response_june.content)
+        july_data = json.loads(response_first_of_july.content)
+        if june_data != {} and july_data != {}:
+            self.assertNotEqual(june_data, july_data)
+
+    def test_chart_third_day_of_month(self):
+        if self.factory is None:
+            return
+        working_reverse = reverse(self.url, kwargs={'domain': 'icds-test', 'step': 'map'})
+        request = self.factory.get(working_reverse)
+        request.user = self.user
+        with mock.patch('custom.icds_reports.views.datetime', LastDayOfJune):
+            response_june = self.view(request, step=u'chart', domain=u'icds-test')
+        with mock.patch('custom.icds_reports.views.datetime', ThirdDayOfJuly):
+            response_first_of_july = self.view(request, step=u'chart', domain=u'icds-test')
+
+        june_data = json.loads(response_june.content)
+        july_data = json.loads(response_first_of_july.content)
+        if june_data != {} and july_data != {}:
+            self.assertNotEqual(june_data, july_data)
+
+    def test_sector_third_day_of_month(self):
+        if self.factory is None:
+            return
+        working_reverse = reverse(self.url, kwargs={'domain': 'icds-test', 'step': 'map'})
+        request = self.factory.get(working_reverse)
+        request.user = self.user
+        with mock.patch('custom.icds_reports.views.datetime', LastDayOfJune):
+            response_june = self.view(request, step=u'chart', domain=u'icds-test', location_id=u'b1')
+        with mock.patch('custom.icds_reports.views.datetime', ThirdDayOfJuly):
+            response_first_of_july = self.view(request, step=u'chart', domain=u'icds-test', location_id=u'b1')
+
+        june_data = json.loads(response_june.content)
+        july_data = json.loads(response_first_of_july.content)
+        if june_data != {} and july_data != {}:
```

Comment:
If this condition doesn't pass, then nothing is tested. Why did you need to add this condition in? If it's necessary, I think it would be better to explicitly declare not to run this test, by setting a class property on one of the inheriting classes. Then at the top of this function you could have:

```python
if not self.run_july_third_test:
    return
```